src/client/entity-editor/common/submit-button.js

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2016  Ben Ockmore
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import {Button} from 'react-bootstrap';
+
+import React from 'react';
+import {connect} from 'react-redux';
+import request from 'superagent-bluebird-promise';
+import {setSubmitError} from '../actions';
+
+function SubmitButton({
+	error,
+	...rest
+}) {
+	const errorElement = error ?
+		<span>{error}</span> :
+		null;
+
+	return (
+		<div>
+			<Button bsStyle="success" {...rest}>
+				Submit
+			</Button>
+			{errorElement}
+		</div>
+	);
+}
+SubmitButton.displayName = 'SubmitButton';
+SubmitButton.propTypes = {
+	error: React.PropTypes.string
+};
+
+function submit(url, data) {
```

Comment:
A lot of this behavior seems like it will be shared by all of our forms. The basic pattern of "check authentication, log in if lacking, continue". Seems like factoring this out would be good. We may also want to modalize login. All this is outside the scope of this PR, though.